orderbook/openapi.yml

```diff
@@ -417,6 +417,40 @@ paths:
                   $ref: "#/components/schemas/Order"
         400:
           description: Problem with parameters like limit being too large.
+  /api/v1/quote:
+    post:
+      summary: Quotes a price and fee for the specified order parameters.
+      description: |
+        This API endpoint accepts a partial order and computes the minimum fee and
+        a price estimate for the order. It returns a full order that can be used
+        directly for signing, and with an included signature, passed directly to
+        the order creation endpoint.
+      requestBody:
+        description: The order parameters to compute a quote for.
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/OrderQuoteRequest"
+      responses:
+        200:
+          description: Quoted order.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/OrderQuote"
+        400:
+          description: Error quoting order.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/FeeAndQuoteError"
+        403:
+          description: Forbidden, your account is deny-listed
+        429:
+          description: Too many order placements
+        500:
+          description: Error quoting an order
```

Comment:
I can't see the difference here between 400 and 500 errors - both are "Error quoting order". 

Also, is the pattern for `#/components/schemas/FeeAndQuoteError` defined here, because it doesn't appear to be.

Reply:
Might be cute to encapsulate all these different specifics into the single FeeAndQuoteError which i imagine is the intention here. Something like 

```rust
pub enum PostQuoteResult {
    QuoteConstructed(OrderQuote),
    FeeAndQuoteError(FeeError),
    Forbidden,
}

pub enum FeeError {
    UnsupportedToken(H160),
    TransferEthToContract,
    SameBuyAndSellToken,
    UnsupportedBuyTokenDestination(BuyTokenDestination),
    UnsupportedSellTokenSource(SellTokenSource),
    ZeroAmount,
}
```

Reply:
`#/components/schemas/FeeAndQuoteError` already exists (ans is used by the `feeAndQuote` route). Since the errors are the same, I just reused the error name for now.

Additionally, we would probably reuse the error type from the existing `feeAndQuote` route.

Reply:
That sounds reasonable, but what about all these other responses that seem out of place

```
        403:
          description: Forbidden, your account is deny-listed
        429:
          description: Too many order placements
        500:
          description: Error quoting an order
```

as I see it, the only things we will need here are 

```
#[derive(Serialize)]
pub enum OrderQuoteResult {
    QuoteConstructed(OrderQuote),
    FeeAndQuoteError(FeeAndQuoteError),
}
```
along with some possible rate limiting...

Reply:
> along with some possible rate limiting...

That is what the 429 is for AFIAU:
```
        429:
          description: Too many order placements # i.e you quoted too many orders and are being rate limited. We should `s/placements/quotes/` to make the error a bit clearer.
```

Additionally, `500` is used as a catch all "internal server error" - i.e. something unexpected went wrong.

Finally:
```
403:
          description: Forbidden, your account is deny-listed
```

Since the `from` is passed in, we can also check that the owner is forbidden by address (we have an "owner deny list" already).

Reply:
Fixed 429 description and added "unexpected" to 500 description.

Reply:
There appears to be something missing in the description of 429 here in your comment...

```
We should `s/placements/quotes/` to make the error a bit clearer.
```

Reply:
It was changed to:
```
        429:
          description: Too many order quotes
```